diff --git a/src/storage/mysql_table_set.cpp b/src/storage/mysql_table_set.cpp
index 3475d3f..599922d 100644
--- a/src/storage/mysql_table_set.cpp
+++ b/src/storage/mysql_table_set.cpp
@@ -11,6 +11,8 @@
 #include "storage/mysql_schema_entry.hpp"
 #include "duckdb/parser/parser.hpp"
 
+#include <sstream>
+
 namespace duckdb {
 
 MySQLTableSet::MySQLTableSet(MySQLSchemaEntry &schema) : MySQLInSchemaSet(schema) {
