diff --git a/spatial/src/spatial/core/index/rtree/rtree_index_plan_scan.cpp b/spatial/src/spatial/core/index/rtree/rtree_index_plan_scan.cpp
index 7a531b5b89f..657d866c221 100644
--- a/spatial/src/spatial/core/index/rtree/rtree_index_plan_scan.cpp
+++ b/spatial/src/spatial/core/index/rtree/rtree_index_plan_scan.cpp
@@ -4,6 +4,7 @@
 #include "duckdb/optimizer/matcher/function_matcher.hpp"
 #include "duckdb/optimizer/optimizer_extension.hpp"
 #include "duckdb/optimizer/remove_unused_columns.hpp"
+#include "duckdb/parser/expression_iterator.hpp"
 #include "duckdb/planner/expression/bound_constant_expression.hpp"
 #include "duckdb/planner/expression/bound_function_expression.hpp"
 #include "duckdb/planner/expression/bound_reference_expression.hpp"
@@ -54,7 +55,7 @@ public:
 			// column id not found in bound columns in the LogicalGet: rewrite not possible
 			rewrite_possible = false;
 		}
-		ExpressionIterator::EnumerateChildren(
+		duckdb::ExpressionIterator::EnumerateChildren(
 			expr, [&](Expression &child) { RewriteIndexExpression(index, get, child, rewrite_possible); });
 	}
 
@@ -255,4 +256,4 @@ void RTreeModule::RegisterIndexPlanScan(DatabaseInstance &db) {
 
 } // namespace core
 
-} // namespace spatial
\ No newline at end of file
+} // namespace spatial
