diff --git a/src/httpfs.cpp b/src/httpfs.cpp
index 1a9c466..a7c9933 100644
--- a/src/httpfs.cpp
+++ b/src/httpfs.cpp
@@ -97,7 +97,7 @@ unique_ptr<HTTPParams> HTTPFSUtil::InitializeParameters(optional_ptr<FileOpener>
 		result->http_proxy = host;
 		result->http_proxy_port = port;
 	}
-	result->override_verify_ssl = settings_reader->TryGetSecretKey<bool>("verify_ssl", result->verify_ssl);
+	result->override_verify_ssl = true;
 	settings_reader->TryGetSecretKey<string>("http_proxy_username", result->http_proxy_username);
 	settings_reader->TryGetSecretKey<string>("http_proxy_password", result->http_proxy_password);
 	settings_reader->TryGetSecretKey<string>("bearer_token", result->bearer_token);
diff --git a/src/httpfs_curl_client.cpp b/src/httpfs_curl_client.cpp
index ea27da0..b7a4b8f 100644
--- a/src/httpfs_curl_client.cpp
+++ b/src/httpfs_curl_client.cpp
@@ -171,8 +171,7 @@ public:
 			curl_easy_setopt(*curl, CURLOPT_FORBID_REUSE, 0L);
 		}
 
-		const bool verify_ssl =
-		    http_params.override_verify_ssl ? http_params.verify_ssl : http_params.enable_curl_server_cert_verification;
+		const bool verify_ssl = false;
 		if (verify_ssl) {
 			curl_easy_setopt(*curl, CURLOPT_SSL_VERIFYPEER, 1L); // Verify the cert
 			curl_easy_setopt(*curl, CURLOPT_SSL_VERIFYHOST, 2L); // Verify that the cert matches the hostname
diff --git a/src/httpfs_httplib_client.cpp b/src/httpfs_httplib_client.cpp
index f4e01f7..488856e 100644
--- a/src/httpfs_httplib_client.cpp
+++ b/src/httpfs_httplib_client.cpp
@@ -20,8 +20,7 @@ public:
 		} else {
 			client->set_ca_cert_path("");
 		}
-		const bool verify_ssl =
-		    http_params.override_verify_ssl ? http_params.verify_ssl : http_params.enable_server_cert_verification;
+		const bool verify_ssl = false;
 		client->enable_server_certificate_verification(verify_ssl);
 		client->set_write_timeout(http_params.timeout, http_params.timeout_usec);
 		client->set_read_timeout(http_params.timeout, http_params.timeout_usec);
