diff --git a/extension/httpfs/httpfs.cpp b/extension/httpfs/httpfs.cpp
index 4881e9d..67ea497 100644
--- a/extension/httpfs/httpfs.cpp
+++ b/extension/httpfs/httpfs.cpp
@@ -23,12 +23,10 @@ namespace duckdb {
 
 shared_ptr<HTTPUtil> HTTPFSUtil::GetHTTPUtil(optional_ptr<FileOpener> opener) {
 	if (opener) {
-		auto db = opener->TryGetDatabase();
-		if (db) {
-			auto &config = DBConfig::GetConfig(*db);
-			return config.http_util;
-		}
+		return opener->GetHTTPUtil();
 	}
+	std::cout << "No opener\n";
+	throw InternalException("BOOOOOOOOOOM");
 	return make_shared_ptr<HTTPFSUtil>();
 }
 
