diff --git a/src/httpfs.cpp b/src/httpfs.cpp
index b7252fde1a3..748149f3926 100644
--- a/src/httpfs.cpp
+++ b/src/httpfs.cpp
@@ -100,6 +100,7 @@ unique_ptr<HTTPParams> HTTPFSUtil::InitializeParameters(optional_ptr<FileOpener>
 	settings_reader->TryGetSecretKey<string>("http_proxy_username", result->http_proxy_username);
 	settings_reader->TryGetSecretKey<string>("http_proxy_password", result->http_proxy_password);
 	settings_reader->TryGetSecretKey<string>("bearer_token", result->bearer_token);
+	result->use_ssl_is_set = settings_reader->TryGetSecretKey<bool>("use_ssl", result->use_ssl);
 
 	Value extra_headers;
 	if (settings_reader->TryGetSecretKey("extra_http_headers", extra_headers)) {
diff --git a/src/httpfs_curl_client.cpp b/src/httpfs_curl_client.cpp
index 2771960628d..01421b83bcf 100644
--- a/src/httpfs_curl_client.cpp
+++ b/src/httpfs_curl_client.cpp
@@ -159,7 +159,9 @@ public:
 			curl_easy_setopt(*curl, CURLOPT_FORBID_REUSE, 0L);
 		}
 
-		if (http_params.enable_curl_server_cert_verification) {
+		const bool use_ssl = http_params.use_ssl_is_set ? http_params.use_ssl : http_params.enable_curl_server_cert_verification;
+
+		if (use_ssl) {
 			curl_easy_setopt(*curl, CURLOPT_SSL_VERIFYPEER, 1L); // Verify the cert
 			curl_easy_setopt(*curl, CURLOPT_SSL_VERIFYHOST, 2L); // Verify that the cert matches the hostname
 		} else {
diff --git a/src/httpfs_httplib_client.cpp b/src/httpfs_httplib_client.cpp
index 36593af7120..4d2c0762d9e 100644
--- a/src/httpfs_httplib_client.cpp
+++ b/src/httpfs_httplib_client.cpp
@@ -18,7 +18,8 @@ public:
 		if (!http_params.ca_cert_file.empty()) {
 			client->set_ca_cert_path(http_params.ca_cert_file.c_str());
 		}
-		client->enable_server_certificate_verification(http_params.enable_server_cert_verification);
+		const bool use_ssl = http_params.use_ssl_is_set ? http_params.use_ssl : http_params.enable_server_cert_verification;
+		client->enable_server_certificate_verification(use_ssl);
 		client->set_write_timeout(http_params.timeout, http_params.timeout_usec);
 		client->set_read_timeout(http_params.timeout, http_params.timeout_usec);
 		client->set_connection_timeout(http_params.timeout, http_params.timeout_usec);
